napalm_panos/panos.py

```diff
@@ -0,0 +1,326 @@
+        return self
+
+    def __exit__(self, exc_type, exc_value, exc_traceback):
+        self.close()
+        if exc_type is not None:
+            self.__raise_clean_exception(exc_type, exc_value, exc_traceback)
+
+    @staticmethod
+    def __raise_clean_exception(exc_type, exc_value, exc_traceback):
+        if exc_type.__name__ not in dir(napalm_base.exceptions) and \
+                        exc_type.__name__ not in __builtins__.keys():
+            epilog = ("NAPALM didn't catch this exception. Please, fill a bugfix on "
+                      "https://github.com/napalm-automation/napalm/issues\n"
+                      "Don't forget to include this traceback.")
+            print(epilog)
+        raise exc_type, exc_value, exc_traceback
+
+    def open(self):
+        self.device = pan.xapi.PanXapi(hostname=self.hostname,
+                                       api_username=self.username,
+                                       api_password=self.password)
+
+    def _open_ssh(self):
+        self.connected = ConnectHandler(device_type='paloalto_panos',
+                                        ip=self.hostname,
+                                        username=self.username,
+                                        password=self.password)
+
+    def close(self):
+        self.device = None
+        self.connected.disconnect()
+
+    def load_template(self, template_name, **template_vars):
+        return napalm_base.helpers.load_template(self, template_name, **template_vars)
+
+    def _import_file(self, filename):
+        key = self.device.keygen()
+
+        params = {
+            'type': 'import',
+            'category': 'configuration',
+            'key': key
+        }
+
+        path = os.path.basename(filename)
+
+        mef = requests_toolbelt.MultipartEncoder(
+            fields={
+                'file': (path, open(filename, 'rb'), 'application/octet-stream')
+            }
+        )
+
+        requests.packages.urllib3.disable_warnings(InsecureRequestWarning)
+        url = 'https://{0}/api/'.format(self.hostname)
+        request = requests.post(
+            url,
+            verify=False,
+            params=params,
+            headers={'Content-Type': mef.content_type},
+            data=mef
+        )
+
+        # if something goes wrong just raise an exception
+        request.raise_for_status()
+
+        response = xml.etree.ElementTree.fromstring(request.content)
+        if response.attrib['status'] == 'error':
+            print "Error while trying to move the config file to the device."
+
+    def load_replace_candidate(self, filename=None, config=None):
+        if config:
+            raise ReplaceConfigException("This method requires a config file.")
+
+        elif filename:
+            if self.loaded is False:
+                if filename.endswith('.xml') is False:
+                    raise MergeConfigException('File must be in XML format.')
+                elif self._save_backup() is False:
```

Comment:
Are these two `if` mutually exclusive? Shouldn't those be just regular `if`s? I understand that by the nature of the algorithm the `if...elif...` here works but they are supposed to be two independent checks, right?